ext/bcmath/libbcmath/src/compare.c

```diff
@@ -45,6 +45,14 @@ bcmath_compare_result _bc_do_compare(bc_num n1, bc_num n2, size_t scale, bool us
 
 	/* First, compare signs. */
 	if (use_sign && n1->n_sign != n2->n_sign) {
+		if (n1->n_len == 1 && n2->n_len == 1 &&
```

Comment:
I understand the change in `bc_str2num`.
However, I don't fully understand this change: why is this necessary on 8.4 but not on lower branches?

Reply:
This is for `Number` objects, so it is not necessary for lower branches.

(`scale` and `n->n_scale` differ only in `Number` objects.)

Perhaps it would be more efficient to compare the `scale` first in the order of the `if` conditions?

Reply:
I don't think this maters a lot for performance.
I missed that the scales are not the same for Number objects of course. Please add a small comment to the check to explain this.